src/Synetic/JsonApiClient/ResponseParser.php

```diff
@@ -16,18 +16,18 @@ class ResponseParser implements ResponseParserInterface {
    * {@inheritdoc}
    */
   public function responseHasErrors(Response $response) {
-    if ($response->getStatusCode() !== 200) {
-      return true;
+    if ($response->getStatusCode() < 200 || $response->getStatusCode() > 299) {
+      return TRUE;
     }
 
     $response->getBody()->rewind();
-    $body = json_decode($response->getBody()->getContents(), true);
+    $body = json_decode($response->getBody()->getContents(), TRUE);
 
-    if (array_key_exists('errors', $body)) {
-      return true;
+    if (is_array($body) && array_key_exists('errors', $body)) {
```

Comment:
Does this give errors without the is_array check? Otherwise you can remove that as the default return state already is false.

Reply:
Yes, for example in case of an 204 (no content) $body isn't an array. So it will produce an error, as array_key_exists expect an array.